src/components/schedule-visualizer/tests/robot-trajectory.test.tsx

```diff
@@ -1,33 +1,128 @@
 import { createMount } from '@material-ui/core/test-utils';
 import React from 'react';
+
 import FakeTrajectoryManager from '../../../mock/fake-traj-manager';
-import RobotTrajectory from '../robot-trajectory';
+import RobotTrajectory, { RobotTrajectoryProps } from '../robot-trajectory';
+import { Trajectory, Conflict } from '../../../robot-trajectory-manager';
+import { TrajectoryColor, TrajectoryDiameter, defaultSettings, Settings } from '../../../settings';
+import { SettingsContext } from '../../app-contexts';
 
 const mount = createMount();
 
-it('renders without crashing', async () => {
-  const trajMgr = new FakeTrajectoryManager();
-  const trajectoryData = await trajMgr.latestTrajectory({
-    request: 'trajectory',
-    param: {
-      map_name: 'L1',
-      duration: 1000,
-      trim: true,
-    },
-  });
-  const trajectoryValue = trajectoryData.values[0];
-  const trajectoryConflict = trajectoryData.conflicts;
-
-  const root = mount(
+const createWrapper = (
+  Component: React.ForwardRefExoticComponent<RobotTrajectoryProps>,
+  trajectory: Trajectory,
+  trajectoryConflict: Conflict[],
+  footprint: number,
+  color: string,
+  settings?: Settings,
+) => {
+  return mount(
     <svg>
-      <RobotTrajectory
-        trajectory={trajectoryValue}
-        conflicts={trajectoryConflict}
-        footprint={0.5}
-        color="black"
-      />
+      {settings ? (
+        <SettingsContext.Provider value={settings}>
+          <Component
+            trajectory={trajectory}
+            conflicts={trajectoryConflict}
+            footprint={footprint}
+            color={color}
+          />
+        </SettingsContext.Provider>
+      ) : (
+        <Component
+          trajectory={trajectory}
+          conflicts={trajectoryConflict}
+          footprint={footprint}
+          color={color}
+        />
+      )}
     </svg>,
   );
+};
+
+describe('Robot Trajectory', () => {
+  let trajMgr;
+  let trajectoryData;
+  let trajectoryValue: Trajectory;
+  let trajectoryConflict: Conflict[];
+  const themeColor = '#4caf50';
+  const fixPathSize = 0.4;
```

Comment:
We could export this as a const value in robot-trajectory.tsx and check that the value matches.